src/duckling/game/drawable/tile-block-drawable-drawer.ts

```diff
@@ -0,0 +1,130 @@
+import {RenderTexture, Texture, Sprite, Graphics, DisplayObject, Container, BaseTexture} from 'pixi.js';
+
+import {Vector} from '../../math/vector';
+import {AssetService} from '../../project/asset.service';
+
+import {TileBlockDrawable, getTileWidth} from './tile-block-drawable';
+
+export function drawTileBlockDrawable(tileBlockDrawable : TileBlockDrawable, assetService : AssetService) : DisplayObject {
+    if (!tileBlockDrawable.textureKey) {
+        return null;
+    }
+
+    let baseTexture = assetService.get({key: tileBlockDrawable.textureKey, type: "TexturePNG"});
+    if (!baseTexture) {
+        return null;
+    }
+    
+    let container = new Container();
+    container.addChild(_constructTileBlockSprite(tileBlockDrawable, baseTexture, assetService));
+    return container;
+}
+
+function _constructTileBlockSprite(tileBlockDrawable : TileBlockDrawable, baseTexture : BaseTexture, assetService : AssetService) : DisplayObject {
+    let tileSize = getTileWidth(tileBlockDrawable, assetService);
+    let tiles = { x: tileBlockDrawable.size.x / tileSize, y: tileBlockDrawable.size.y / tileSize};
+    
+    let positions = _getPositions(tiles);
+    let calculateOffset: {x: boolean, y: boolean} = _getOffsets(tiles);
+    
+    return _constructTileBlockContainer(positions, calculateOffset, baseTexture, tileSize, tiles);
+}
+
+function _getPositions(tiles: Vector): Vector[] {
+    let positions: Vector[] = [];
+    
+    let xPositions: number[] = _getAxisPositions(tiles.x);
+    let yPositions: number[] = _getAxisPositions(tiles.y);
+    
+    let position: Vector;
+    for (let xPosition of xPositions) {
```

Comment:
The xPositions and yPositions variables are unnecessary. 
for (let xPosition of _getAxisPositions(tiles.x) {
   for (let yPosition of _getAxisPositions(tiles.y) {